Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,417 @@
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// This tool allows the migration of game object obsolete components into up-to-date versions.
+    /// Requires Implementation of IMigrationhandler.
+    /// </summary>
+    public class MigrationWindow : EditorWindow
+    {
+        private enum ToolbarOption 
+        { 
+            Objects = 0, 
+            Scenes = 1, 
+            Project = 2 
+        };
+
+        private static readonly string[] toolbarTitles =
+{
+            "Objects",
+            "Scenes",
+            "Full Project"
+        };
+
+        private ToolbarOption selectedToolbar = ToolbarOption.Objects;
+        private Vector2 scrollPosition = Vector2.zero;
+
+        private const string DependencyWindowURL = "https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Tools/MigrationWindow.html";
+        private const string WindowTitle = "Migration Window";
+        private const string WindowDescription = "This tool allows the migration of obsolete components into up-to-date versions. Component-specific implementation of the MigrationHandler Interface is required.";
+
+        private const string SceneExtension = ".unity";
+        private const string PrefabExtension = ".prefab";
+
+        private Dictionary<string, Type> migrationTypesMap = new Dictionary<string, Type>();
+        private String[] migrationTypeNames;
+        private int selectedMigrationHandlerIndex;
+        private List<Object> migrationObjects = new List<Object>();
+        private Object selection;
+
+        private dynamic migrationTypeInstance
+        {
+            get
+            {
+                var selectedMigrationTypeName = migrationTypeNames[selectedMigrationHandlerIndex];
+                var migrationHandlerType = migrationTypesMap[selectedMigrationTypeName];
+                return Activator.CreateInstance(migrationHandlerType) as IMigrationHandler;
+            }
+        }
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Migration Window", false, 4)]
+        private static void ShowWindow()
+        {
+            var window = GetWindow<MigrationWindow>(typeof(SceneView));
+            window.titleContent = new GUIContent(WindowTitle, EditorGUIUtility.IconContent("d_TimelineEditModeRippleOFF").image);
+            window.minSize = new Vector2(400.0f, 300.0f);
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            if (EditorApplication.isPlaying || EditorApplication.isPaused)
+            {
+                GUI.enabled = false;
+            }
+            DrawHeader();
+            DrawMigrationTypeSelector();
+            DrawMigrationToolbars();
+        }
+
+        private void OnFocus()
+        {
+            RefreshAvailableTypes();
+        }
+
+        private void RefreshAvailableTypes()
+        {
+            var type = typeof(IMigrationHandler);
+            var types = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
Note that we've historically seen cases where GetTypes would throw a ton of exceptions (i.e. if there are assemblies that aren't loadable).

See https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5949/files for reference.

I would actually probably lean on using GetLoadableTypes as is done in that PR to avoid hitting that type of issue going forward.

Do I know exactly what scenario? Where we observed it, it was in some other development envs where there were some other interesting non-Unity standard setups AFAIK but I could imagine that someone who is pulling in binaries (i.e. dragging in a DLL to their project) could probably trigger that as well.

Reply:
Also can look at TypeExtensions.GetAllSubClassesOf() option. I think for your purposes that should work here?

Reply:
Changed to GetLoadableTypes as an instance is actually needed.